iml-wasm-components/iml-stratagem/src/inode_table.rs

```diff
@@ -0,0 +1,371 @@
+    .controller(|controller| request_controller = Some(controller))
+    .fetch_json(Msg::InodesFetched);
+
+    (fut, request_controller)
+}
+
+fn get_inode_elements<T>(inodes: &Vec<INode>) -> Vec<El<T>> {
+    inodes
+        .into_iter()
+        .map(|x| tr![td![x.counter_name], td![x.count.to_string()]])
+        .collect()
+}
+
+/// View
+pub fn view(model: &Model) -> El<Msg> {
+    if model.destroyed {
+        seed::empty()
+    } else {
+        let inodes = get_inode_elements(&model.inodes);
+
+        if !inodes.is_empty() {
+            table(inodes)
+        } else {
+            seed::empty()
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use futures::sync::{oneshot, oneshot::Sender};
+    use seed::fetch::{Request, ResponseWithDataResult, Status, StatusCategory};
+    use std::sync::{Arc, Mutex};
+    use wasm_bindgen_test::wasm_bindgen_test_configure;
+
+    wasm_bindgen_test_configure!(run_in_browser);
+
+    use wasm_bindgen_test::*;
+
+    fn destroy_after_delay() -> impl Future<Item = Msg, Error = Msg> {
+        iml_sleep::Sleep::new(1000)
+            .map(move |_| Msg::Destroy)
+            .map_err(|_| unreachable!())
+    }
+
+    #[wasm_bindgen_test(async)]
+    pub fn test_inodes_with_data() -> impl Future<Item = (), Error = JsValue> {
+        #[derive(Debug)]
+        pub struct TestModel {
+            model: Model,
+            p: Arc<
+                Mutex<
+                    Option<
+                        Sender<(
+                            Option<String>,
+                            Option<String>,
+                            Option<String>,
+                            Option<String>,
+                        )>,
+                    >,
+                >,
+            >,
+        }
+
+        pub fn assert_view(TestModel { p, model }: &TestModel) -> El<Msg> {
+            let el = view(&model);
+
+            if !el.children.is_empty() {
+                let tr1 = el.children[0].clone();
+                let td11 = tr1.children[0].children[0].clone().text;
+                let td12 = tr1.children[1].children[0].clone().text;
+                let tr2 = el.children[1].clone();
+                let td21 = tr2.children[0].children[0].clone().text;
+                let td22 = tr2.children[1].children[0].clone().text;
+
+                p.lock()
+                    .unwrap()
+                    .take()
+                    .map(|p| p.send((td11, td12, td21, td22)));
+            }
+
+            el
+        }
+
+        pub fn test_update(msg: Msg, model: &mut TestModel, orders: &mut Orders<Msg>) {
+            update(msg, &mut model.model, orders);
+            orders.perform_cmd(destroy_after_delay());
+        }
+
+        pub fn render() -> impl Future<Item = (), Error = JsValue> {
+            let (p, c) = oneshot::channel::<(
```

Comment:
Instead of sending a tuple of strings, what about sending the element?

Reply:
Works fine. Updated.